dask/dataframe/io/parquet/arrow.py

```diff
@@ -342,14 +342,24 @@ def read_partition(
                 open_file_func=partial(fs.open, mode="rb"),
             )
 
+        # Ensure `columns` and `partitions` do not overlap
+        columns_and_parts = columns.copy()
+        if columns_and_parts and partitions:
+            for part_name in partitions.partition_names:
+                if part_name in columns:
+                    columns.remove(part_name)
+                else:
+                    columns_and_parts.append(part_name)
+            columns = columns or None
```

Comment:
Is this only for the case where `part_name` completely overlaps with `columns`, `columns` is removed?

Reply:
Right - If we get to this point, all items in `columns` exist in `partitions.partition_names`.  Therefore, `columns` will become an empty list.  However, if we pass `columns=[]` as an argument to `piece.read`, we will get back an empty dataframe.  To avoid this, we now pass `columns=None` to read all columns for this special case. 

Note that there may be a better (more performant) way to handle the case that the user only want to read the partition columns.  Since we do not actually want/nmeed to read any of the columns that are actually written to the partquet files, we can probably use the metadata to populate the partition directly.